core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -550,7 +547,14 @@ func (r *logRecoverer) tryExpire(ctx context.Context, ids ...string) (int, error
 			if logBlock := rec.payload.Trigger.LogTriggerExtension.BlockNumber; int64(logBlock) < start {
 				// we can't recover this log anymore, so we remove it from the visited list
 				lggr.Debugw("removing expired log: old block", "upkeepID", rec.payload.UpkeepID,
-					"logBlock", logBlock, "start", start)
+					"latestBlock", latestBlock, "logBlock", logBlock, "start", start)
+				pending := make([]ocr2keepers.UpkeepPayload, 0, len(r.pending))
+				for _, p := range r.pending {
+					if p.WorkID != rec.payload.WorkID {
+						pending = append(pending, p)
+					}
+				}
+				r.pending = pending
```

Comment:
not able to comment on line 562, but can that result in appending duplicate payload to pending?